api/v1alpha1/flowcollector_types.go

```diff
@@ -23,11 +23,39 @@ import (
 
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-const (
-	AgentIPFIX = "ipfix"
-	AgentEBPF  = "ebpf"
+// AgentType selects the Agent technology that is used to trace the system flows.
+// +kubebuilder:validation:Enum:=IPFIX;EBPF
+type AgentType string
+
+var (
+	AgentIPFIX AgentType = "IPFIX"
+	AgentEBPF  AgentType = "EBPF"
 )
 
+// AgentConfig is a discriminated union that allows to select either ipfix or ebpf, but does not
+// allow defining both fields.
+// +union
+type AgentConfig struct {
+	// Select the flows tracing agent. Possible values are "IPFIX" (default) to use
+	// the IPFIX collector, or "EBPF" to use NetObserv eBPF agent. When using IPFIX with OVN-Kubernetes
+	// CNI, NetObserv will configure OVN's IPFIX exporter. Other CNIs are not supported, they could
+	// work but require manual configuration.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="IPFIX";"EBPF"
+	// +kubebuilder:validation:Required
+	Type AgentType `json:"type"`
+
+	// Settings related to IPFIX-based flow reporter when the "agent" property is set
+	// to "ipfix".
+	// +optional
+	IPFIX FlowCollectorIPFIX `json:"ipfix,omitempty"`
```

Comment:
Discriminated union documentations state that both IPFIX and EBPF fields should be optional pointers. However, I had to set them as values because otherwise we would loose all the default values and users would be forced to define either ipfix of ebpf sections.

https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#discriminated-unions